### ISSUE TITLE
Update googlefonts.php

### ISSUE THREAD

**Issue by razor2016** [reduxframework/redux-framework#3742]
Hi,

Can you please keep updating the `googlefonts.php` file (`path: wp-content\plugins\redux-framework\redux-core\inc\fields\typography\googlefonts.php`)?

I am constructing my own font url instead of the redux generated one. Why? Because, I need to insert the font url into the add_editor_style() function for the fonts to be available in the block editor and I could not get redux generated url outside of the plugin.

What am I doing with that particular file? 

I am using that file to check if the font variants are indeed available. Let's say admin user sets lato with 400 regular weight for body font. In this case, I also need 700 variant for bolder fonts. So I am checking if 600 is available. If not, then check if 700 is available. Then finally I am pushing valid font variants into `https://fonts.googleapis.com/` to construct my url.

I hope you understand this. Please let me know.

Thanks

**Comment by dovy** [comment]
So have you tried the new Redux (version 4)? It has Google Font updates built in. We worked to resolve this on a more regular basis.

**Comment by razor2016** [comment]
Yes, I am running on v4. And I know you have google fonts built on. New fonts are available in the redux option panel. 

Let me try to explain:

My goal is to construct a google font url like this: `https://fonts.googleapis.com/css?family=Nunito+Sans%3A300%2C400%2C600%2C700%2C300italic%2C400italic%2C600italic%2C700italic%7CQuicksand%3A700&#038;ver=1.0.4#038;subset`

To achieve it, I get the fonts values from the option panel and then I need to check whether some font variants (300, 400, 600 etc) for a particular font are valid or not. How do I do that?

I am currently using `googlefonts.php` for the checking. I updated the plugin and saw that this file is outdated.

Basically, I need all the valid fonts data in a json format.

Thanks

**Comment by STV11C** [comment]
That’s a pretty nice idea. I am having the same issue where additional font weights are not being loaded for Bold, Italic etc. 

I was going to get around it by adding a dedicated Redux option for a Bold font etc.

Eg. A user selects the Body font - say Roboto Regular - and the Heading fonts as Lato Bold - then the Roboto Bold weight is not loaded. 

So I may include a Body Bold and Body Italic Option so the user can specify the weight.


Regards,
Stephan

**Comment by dovy** [comment]
We could expand the weights pretty easy. Haha.

**Comment by STV11C** [comment]
That could work :) maybe instead of a single dropdown for Weight, it could be a multi select.

A validation check could be that if the same Font/Weight combo is selected, it only loads it once.

Regards,
Stephan

**Comment by dovy** [comment]
That could work. Anyone want to take it on? I'm still putting out fires, hah.

**Comment by dovy** [comment]
P.S. Here's how we get all the current Google fonts, even those updated.

https://github.com/reduxframework/redux-framework-4/blob/master/redux-core/inc/fields/typography/class-redux-typography.php#L1269-L1273

**Comment by razor2016** [comment]
> P.S. Here's how we get all the current Google fonts, even those updated.
> 
> https://github.com/reduxframework/redux-framework-4/blob/master/redux-core/inc/fields/typography/class-redux-typography.php#L1269-L1273

Ah, this is great!! I'm closing this issue.

**Comment by STV11C** [comment]
I’ll give it a crack 

Regards,
Stephan
